src/synthesizer/sed.py

```diff
@@ -1314,18 +1314,25 @@ def calculate_ionising_photon_production_rate(
         x = self._lam
         y = (llam * self.lam / h.to(erg / Hz) / c.to(angstrom / s)).value
 
-        # Get value of luminosity at ionisation wavelength
-        ionisation_y = np.interp(
-            ionisation_wavelength.to(angstrom).value, x, y
-        )
-
         # Restrict arrays to ionisation regime
         x = x[ionisation_mask]
-        y = y[ionisation_mask]
+        if len(y.shape) == 1:
+            y = y[ionisation_mask]
+        else:
+            y = y[:, ionisation_mask]
+
+        # Add a final data point at the ionising energy to ensure full
+        # coverage.
+        x0 = ionisation_wavelength.to(angstrom).value
+        if len(y.shape) == 1:
```

Comment:
Same with this block, it'll fail for 3D arrays. You could use scipy and `interp1d` with the axis argument or I think you could just use pure numpy with `apply_along_axis`
```
y0 = np.apply_along_axis(lambda y_: np.interp(x0, x, y_), axis=-1, arr=y)
``` 
https://numpy.org/doc/2.0/reference/generated/numpy.apply_along_axis.html

Reply:
Fixed. See comment below.